---
title: "Ancestral State Reconstruction for ST 26"
output: html_document
---

# Step 0: Install ggtree and phytools, if necessary

```{r}
### install ggtree, if necessary (uncomment lines in code chunk below which state with a single # character)

#if (!requireNamespace("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")

#BiocManager::install("ggtree")

### install phytools, if necessary

#install.packages("phytools")
```

# Step 1: Load ggtree and phytools

```{r}
### load ggtree
library(ggtree)
### load phytools
library(phytools)
```

# Step 2: Read in ST 26 phylogeny, constructed using BEAST 2, LogCombiner-2, and TreeAnnotator-2

```{r}
### read BEAST 2 tree in NEXUS format

paranthracis <- read.beast(file = "snippy_TVMef5gamma_relaxed_bdsky_10burninLC_0burninTS_MCC_CAnodes.tree")
```

# Step 3: Create a vector corresponding to the binary discrete state of each isolate (*ces*-positive or *ces*-negative), named with the according tree tip labels

```{r}
### construct vector from tip label names

ces <- as.factor(unlist(lapply(strsplit(x = paranthracis@phylo$tip.label, split = "_"), "[[", 6)))

### name vector according to associated tree tip labels

names(ces) <- paranthracis@phylo$tip.label
```

# Step 4: Simulate stochastic character maps on the phylogeny

## These commands use the make.simmap function, the all-rates-different model, 10,000 simulations of stochastic character  and one of the following three priors on the root node, corresponding to the *ces*-positive and *ces*-negative state of the root node:

### 1. Equal probability of the root node belonging to a *ces*-positive or *ces*-negative state

### 2. Estimated probabilities of the root node belonging to a *ces*-positive or *ces*-negative state, obtained using the make.simmap function

### 3. Probability of the root node being in a *ces*-positive or *ces*-negative state set to 0.2 and 0.8, respectively, as the probability of the ST 26 ancestor being *ces*-negative was estimated to be between 0.78 and 0.82 (depending on the choice of outgroup) when core SNPs among all Group III *B. cereus s.l.* genomes were used for ancestral state reconstruction 


```{r}
### 1. Equal probability of ces-negative and ces-positive

mtrees <- make.simmap(tree = paranthracis@phylo, x = ces, model = "ARD", nsim = 10000, Q="empirical")

### 2. Probability of root node being in ces-positive or ces-negative state is estimated using make.simmap

mtrees <- make.simmap(tree = paranthracis@phylo, x = ces, model = "ARD", nsim = 10000, Q="empirical", pi = "estimated")

### 3. Probability of being in ces-negative or ces-positive state is set to 0.8 and 0.2, respectively

mtrees <- make.simmap(tree = paranthracis@phylo, x = ces, model = "ARD", nsim = 10000, Q="empirical", pi = c(0.8, 0.2))
```

# Step 5: Create a summary of the simulated trees

```{r}
### Summarize trees

pd <- summary(mtrees)
```

# Step 6: Plot a tree with the posterior density for a mapped character from the stochastic character mapping on the tree performed above

```{r}
### Plot tree

cols<-setNames(c("lightseagreen","violetred3"),levels(ces))
obj <- densityMap(mtrees, states=levels(ces)[2:1], plot=F)
n <- length(obj$cols)
obj$cols[1:n]<-colorRampPalette(rev(cols), space="Lab")(n)
plot(obj, fsize=c(0.4,1), color=cols)
add.scale.bar(x = 0, y = 3)

### uncomment the following line to add node labels corresponding to posterior probability of being in a ces-negative state

#nodelabels(text = pd$ace[,1], frame = "none", cex = 0.7, adj = -0.1)

### uncomment the follwing line to add pie chart node labels corresponding to posterior probability of being in a particular state

nodelabels(pie = pd$ace, piecol = cols, cex= 0.5)
```

