# Group_III_bacillus_cereus
Supplementary material for the 2019-2020 cereulide-producing Group III *B. cereus s.l.* phylogenomics study

## Phylogenies

### GroupIII/ksnp3_outgroupAmes_core_SNPs_matrix.fasta.treefile

Maximum likelihood phylogeny constructed using core SNPs identified among 71 emetic Group III *B. cereus s.l.* genomes 
and their closely related, non-emetic counterparts, plus outgroup genome *B. anthracis.* str. Ames. Core SNPs were 
identified using kSNP3, and the phylogeny was constructed using IQ-TREE/ModelFinder.

### GroupIII/ksnp3_outgroupAmes_SNPs_in_majority0.5_matrix.fasta.treefile

Maximum likelihood phylogeny constructed using majority SNPs identified among 71 emetic Group III *B. cereus s.l.* genomes 
and their closely related, non-emetic counterparts, plus outgroup genome *B. anthracis.* str. Ames. Majority SNPs were 
identified using kSNP3, and the phylogeny was constructed using IQ-TREE/ModelFinder.

### GroupIII/ksnp3_outgroupGCF_002574215_core_SNPs_matrix.fasta.treefile

Maximum likelihood phylogeny constructed using core SNPs identified among 71 emetic Group III *B. cereus s.l.* genomes 
and their closely related, non-emetic counterparts, plus outgroup genome *B. cereus s.l.* str. AFS057383. Core SNPs were 
identified using kSNP3, and the phylogeny was constructed using IQ-TREE/ModelFinder.

### GroupIII/ksnp3_outgroupGCF_002574215_SNPs_in_majority0.5_matrix.fasta.treefile

Maximum likelihood phylogeny constructed using majority SNPs identified among 71 emetic Group III *B. cereus s.l.* genomes 
and their closely related, non-emetic counterparts, plus outgroup genome *B. cereus s.l.* str. AFS057383. Majority SNPs were 
identified using kSNP3, and the phylogeny was constructed using IQ-TREE/ModelFinder.

### Cluster1234/ksnp3_Cluster1234_outgroupCluster7_core_SNPs_matrix.fasta.treefile

Maximum likelihood phylogeny constructed using core SNPs identified among 33 Group III *B. cereus s.l.* genomes 
assigned to RhierBAPS clusters 1-4, plus a cluster 7 outgroup genome. Core SNPs were identified using kSNP3, and the phylogeny was constructed using IQ-TREE/ModelFinder.

### Cluster1234/ksnp3_Cluster1234_outgroupCluster7_SNPs_in_majority0.5_matrix.fasta.treefile

Maximum likelihood phylogeny constructed using majority SNPs identified among 33 Group III *B. cereus s.l.* genomes 
assigned to RhierBAPS clusters 1-4, plus a cluster 7 outgroup genome. Majority SNPs were identified using kSNP3, and the phylogeny was constructed using IQ-TREE/ModelFinder.

### Cluster1234/ksnp3_Cluster1234_outgroupCluster7_SNPs_all_matrix.fasta.treefile

Maximum likelihood phylogeny constructed using all SNPs identified among 33 Group III *B. cereus s.l.* genomes 
assigned to RhierBAPS clusters 1-4, plus a cluster 7 outgroup genome. All SNPs were identified using kSNP3, and the phylogeny was constructed using IQ-TREE/ModelFinder.

### Cluster123/ksnp3_Cluster123_outgroupCluster4_core_SNPs_matrix.fasta.treefile

Maximum likelihood phylogeny constructed using core SNPs identified among 31 Group III *B. cereus s.l.* genomes 
assigned to RhierBAPS clusters 1-3, plus a cluster 4 outgroup genome. Core SNPs were identified using kSNP3, and the phylogeny was constructed using IQ-TREE/ModelFinder.

### Cluster123/ksnp3_Cluster123_outgroupCluster4_SNPs_in_majority0.5_matrix.fasta.treefile

Maximum likelihood phylogeny constructed using majority SNPs identified among 31 Group III *B. cereus s.l.* genomes 
assigned to RhierBAPS clusters 1-3, plus a cluster 4 outgroup genome. Majority SNPs were identified using kSNP3, and the phylogeny was constructed using IQ-TREE/ModelFinder.

### Cluster123/ksnp3_Cluster123_outgroupCluster4_SNPs_all_matrix.fasta.treefile

Maximum likelihood phylogeny constructed using all SNPs identified among 31 Group III *B. cereus s.l.* genomes 
assigned to RhierBAPS clusters 1-3, plus a cluster 4 outgroup genome. All SNPs were identified using kSNP3, and the phylogeny was constructed using IQ-TREE/ModelFinder.

### ST26/trees/Original2018Select3_relaxed_skyline_10burnin_combined_MEDIANnodes_0burninTA.tree

Rooted, time-scaled maximum clade credibility (MCC) phylogeny constructed using core SNPs identified among 37 Group III *B. cereus s.l.* genomes belonging to sequence type (ST) 26 (i.e., the **Original 2018/Select 3 NYS** isolate set listed in **Supplemental Table S3**). Core SNPs were identified using Snippy version 4.3.6. The phylogeny was constructed using the results of five independent runs using a relaxed lognormal clock model, the Standard_TVMef nucleotide substitution model with 5 gamma categories, and a Coalescent Bayesian Skyline population model implemented in BEAST version 2.5.1, with 10% burn-in applied to each run. LogCombiner-2 was used to combine BEAST 2 log files, and TreeAnnotator-2 was used to construct the phylogeny using median node heights. The phylogeny can be easily loaded into <a href="http://tree.bio.ed.ac.uk/software/figtree/">FigTree</a> to view node ages, node height highest posterior density (HPD) intervals, and other valuable metrics. 

### ST26/trees/Original2018Select3NoEstimated_relaxed_skyline_10burnin_combined_MEDIANnodes_0burninTA.tree

Rooted, time-scaled maximum clade credibility (MCC) phylogeny constructed using core SNPs identified among 33 Group III *B. cereus s.l.* genomes belonging to sequence type (ST) 26 (i.e., the **Original 2018/Select 3 NYS No Estimated** isolate set listed in **Supplemental Table S3**). Core SNPs were identified using Snippy version 4.3.6. The phylogeny was constructed using the results of five independent runs using a relaxed lognormal clock model, the Standard_TVMef nucleotide substitution model with 5 gamma categories, and a Coalescent Bayesian Skyline population model implemented in BEAST version 2.5.1, with 10% burn-in applied to each run. LogCombiner-2 was used to combine BEAST 2 log files, and TreeAnnotator-2 was used to construct the phylogeny using median node heights. The phylogeny can be easily loaded into <a href="http://tree.bio.ed.ac.uk/software/figtree/">FigTree</a> to view node ages, node height highest posterior density (HPD) intervals, and other valuable metrics. 

### ST26/trees/Original2018Select5_relaxed_skyline_10burninLC_MCC_MEDIANnodes_0burninTA.tree

Rooted, time-scaled maximum clade credibility (MCC) phylogeny constructed using core SNPs identified among 39 Group III *B. cereus s.l.* genomes belonging to sequence type (ST) 26 (i.e., the **Original 2018/Select 5 NYS** isolate set listed in **Supplemental Table S3**). Core SNPs were identified using Snippy version 4.3.6. The phylogeny was constructed using the results of five independent runs using a relaxed lognormal clock model, the Standard_TVMef nucleotide substitution model with 5 gamma categories, and a Coalescent Bayesian Skyline population model implemented in BEAST version 2.5.1, with 10% burn-in applied to each run. LogCombiner-2 was used to combine BEAST 2 log files, and TreeAnnotator-2 was used to construct the phylogeny using median node heights. The phylogeny can be easily loaded into <a href="http://tree.bio.ed.ac.uk/software/figtree/">FigTree</a> to view node ages, node height highest posterior density (HPD) intervals, and other valuable metrics. 

### ST26/trees/Original2018Select10_relaxed_skyline_10burninLC_MCC_MEDIANnodes_0burninTA.tree

Rooted, time-scaled maximum clade credibility (MCC) phylogeny constructed using core SNPs identified among 44 Group III *B. cereus s.l.* genomes belonging to sequence type (ST) 26 (i.e., the **Original 2018/Select 10 NYS** isolate set listed in **Supplemental Table S3**). Core SNPs were identified using Snippy version 4.3.6. The phylogeny was constructed using the results of five independent runs using a relaxed lognormal clock model, the Standard_TVMef nucleotide substitution model with 5 gamma categories, and a Coalescent Bayesian Skyline population model implemented in BEAST version 2.5.1, with 10% burn-in applied to each run. LogCombiner-2 was used to combine BEAST 2 log files, and TreeAnnotator-2 was used to construct the phylogeny using median node heights. The phylogeny can be easily loaded into <a href="http://tree.bio.ed.ac.uk/software/figtree/">FigTree</a> to view node ages, node height highest posterior density (HPD) intervals, and other valuable metrics. 

### ST26/trees/Original2018AllNYS_relaxed_skyline_10burnin_MCC_MEDIANnode_10burninLC_0burninTA.tree

Rooted, time-scaled maximum clade credibility (MCC) phylogeny constructed using core SNPs identified among 64 Group III *B. cereus s.l.* genomes belonging to sequence type (ST) 26 (i.e., the **Original 2018/All NYS Outbreak** isolate set listed in **Supplemental Table S3**). Core SNPs were identified using Snippy version 4.3.6. The phylogeny was constructed using the results of five independent runs using a relaxed lognormal clock model, the Standard_TVMef nucleotide substitution model with 5 gamma categories, and a Coalescent Bayesian Skyline population model implemented in BEAST version 2.5.1, with 10% burn-in applied to each run. LogCombiner-2 was used to combine BEAST 2 log files, and TreeAnnotator-2 was used to construct the phylogeny using median node heights. The phylogeny can be easily loaded into <a href="http://tree.bio.ed.ac.uk/software/figtree/">FigTree</a> to view node ages, node height highest posterior density (HPD) intervals, and other valuable metrics. 

### ST26/trees/New2020Select3_relaxed_skyline_10burninLC_MCC_MEDIANnodes_0burninTA.tree

Rooted, time-scaled maximum clade credibility (MCC) phylogeny constructed using core SNPs identified among 46 Group III *B. cereus s.l.* genomes belonging to sequence type (ST) 26 (i.e., the **New 2020/Select 3 NYS** isolate set listed in **Supplemental Table S3**). Core SNPs were identified using Snippy version 4.3.6. The phylogeny was constructed using the results of five independent runs using a relaxed lognormal clock model, the Standard_TVMef nucleotide substitution model with 5 gamma categories, and a Coalescent Bayesian Skyline population model implemented in BEAST version 2.5.1, with 10% burn-in applied to each run. LogCombiner-2 was used to combine BEAST 2 log files, and TreeAnnotator-2 was used to construct the phylogeny using median node heights. The phylogeny can be easily loaded into <a href="http://tree.bio.ed.ac.uk/software/figtree/">FigTree</a> to view node ages, node height highest posterior density (HPD) intervals, and other valuable metrics. 

### ST26/trees/New2020Select5_relaxed_skyline_10burninLC_MCC_MEDIANnodes_0burninTA.tree

Rooted, time-scaled maximum clade credibility (MCC) phylogeny constructed using core SNPs identified among 48 Group III *B. cereus s.l.* genomes belonging to sequence type (ST) 26 (i.e., the **New 2020/Select 5 NYS** isolate set listed in **Supplemental Table S3**). Core SNPs were identified using Snippy version 4.3.6. The phylogeny was constructed using the results of five independent runs using a relaxed lognormal clock model, the Standard_TVM nucleotide substitution model with 5 gamma categories, and a Coalescent Bayesian Skyline population model implemented in BEAST version 2.5.1, with 10% burn-in applied to each run. LogCombiner-2 was used to combine BEAST 2 log files, and TreeAnnotator-2 was used to construct the phylogeny using median node heights. The phylogeny can be easily loaded into <a href="http://tree.bio.ed.ac.uk/software/figtree/">FigTree</a> to view node ages, node height highest posterior density (HPD) intervals, and other valuable metrics.

### ST26/trees/New2020Select5NoEstimated_relaxed_skyline_10burninLC_MCC_MEDIANnodes_0burninTA.tree

Rooted, time-scaled maximum clade credibility (MCC) phylogeny constructed using core SNPs identified among 43 Group III *B. cereus s.l.* genomes belonging to sequence type (ST) 26 (i.e., the **New 2020/Select 5 NYS No Estimated** isolate set listed in **Supplemental Table S3**). Core SNPs were identified using Snippy version 4.3.6. The phylogeny was constructed using the results of five independent runs using a relaxed lognormal clock model, the Standard_TVMef nucleotide substitution model with 5 gamma categories, and a Coalescent Bayesian Skyline population model implemented in BEAST version 2.5.1, with 10% burn-in applied to each run. LogCombiner-2 was used to combine BEAST 2 log files, and TreeAnnotator-2 was used to construct the phylogeny using median node heights. The phylogeny can be easily loaded into <a href="http://tree.bio.ed.ac.uk/software/figtree/">FigTree</a> to view node ages, node height highest posterior density


If you would like to access the raw BEAST 2 log files used to create the ST 26 tree(s), please <a href="https://github.com/lmc297/Group_III_bacillus_cereus/issues">create an issue with your contact information</a> so that we can arrange a way to send them to you (the log files are very large, and sadly cannot be hosted on GitHub!).

## Raw Data

### ST26/xml/\*_relaxed_skyline.xml

The raw BEAST 2 XML files used to construct the phylogenies in ST26/trees. If you would like to access the raw BEAST 2 log files produced by these XMLs, please <a href="https://github.com/lmc297/Group_III_bacillus_cereus/issues">create an issue with your contact information</a> so that we can arrange a way to send them to you (the log files are very large, and sadly cannot be hosted on GitHub!).

## Sample Scripts

### GroupIII/ancestral_state_ML_GroupIII.Rmd

R Markdown document describing how ancestral state reconstruction was performed for 71 de-replicated Group III *B. cereus s.l.* genomes.

### Cluster1234/ancestral_state_Cluster1234_outgroupCluster7.Rmd

R Markdown document describing how ancestral state reconstruction was performed for 33 Group III *B. cereus s.l.* genomes assigned to RhierBAPS clusters 1-4.

### Cluster123/ancestral_state_Cluster123_outgroupCluster4.Rmd

R Markdown document describing how ancestral state reconstruction was performed for 31 Group III *B. cereus s.l.* genomes assigned to RhierBAPS clusters 1-3.

### ST26/scripts/ancestral_state_beast_ST26.Rmd

R Markdown document describing how ancestral state reconstruction was performed for the ST 26 phylogenies constructed using BEAST 2 (available in ST26/trees)

## old_preprint_results.tar.gz

Included for archival purposes only; contains obsolete results associated with a <a href="https://www.biorxiv.org/content/10.1101/2020.05.12.090951v1">preprint</a> for this study. The ST 26 temporal phylogeny results in this compressed directory were obtained using a set of 23 de-replicated ST 26 genomes. While de-replication is a common practice, <a href="https://academic.oup.com/mbe/article/doi/10.1093/molbev/msaa136/5850868">a recent study</a> has shown that the common practice of removing duplicate sequences to reduce a set of genomes to a set of unique sequences can lead to biases when constructing temporal phylogenies using Bayesian methods.
